### ISSUE TITLE
Dropdown label and value being available in plain text variable picker fields

### ISSUE THREAD

**Issue by jamesmacwhite** [verbb/formie#2175]
### What are you trying to do?

When using the variable picker field in context like email notifications, we have a scenario were we want to use a dropdown field but want to select the label and not the value to be displayed in an email notification context.

At the moment the field is available in the picker, but it is defaulting to the value, which is a GUID with no context, given its for an external integration system, where as the label is the front facing friendly value we want to show for human usage.

Would it be possible to allow dropdown fields to provide both the label and value as available to select in a picker?

### What's your proposed solution?

Consideration of having the label and value as options in a picker. Given dropdown fields are a single option currently, I guess compatibility needs to be considered, to not break existing setups.

**Comment by engram-design** [comment]
Yeah, this is on our list to [expand](https://github.com/verbb/formie/issues/2223) the variable picker like this!

In Formie 3, there is a "Email Notification Value" for fields, which might work. But that's set for all instances of using that field in the variable picker.

**Comment by jamesmacwhite** [comment]
Thanks! I checked the field and it appears label is set and default, as I don't remember setting it:

![image](https://github.com/user-attachments/assets/ef217416-5f89-4956-807a-5982cba610aa)

The field value is however displayed on email previews and confirmed to be appearing on the actual notifications:

![image](https://github.com/user-attachments/assets/bafb4f45-2158-414f-9c79-ad54e35aa019)

**Comment by engram-design** [comment]
Hmm, no custom email templates? [This](https://github.com/verbb/formie/blob/abfcfa45f21c2807e1c38097b6ed6cc80c0c14ab/src/base/OptionsField.php#L249-L257) is how that variable is determined.

**Comment by jamesmacwhite** [comment]
Not in this specific scenario, the rest of email notifications use MJML, where you can control fields in Twig, so not an issue, but for this specific form it is using the default email notification field with the fields selected from the picker.

**Comment by jamesmacwhite** [comment]
I did: `dd($this->emailValue === 'label')` in the condition and checked the email preview HTTP request it returns `false` for me, so the condition isn't being met to use the label by the looks of it.

**Comment by jamesmacwhite** [comment]
OK weirdly, saving the form with Value set then switching it to Label now appears to trigger the logic to use the label, potentially the form model/settings out of sync or not setting label as a default.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.1](https://github.com/verbb/formie/releases/tag/4.0.0-beta.1).
